### PR TITLE
Fragmentswork

### PR DESCRIPTION
**Pull request by deliasgomez**
WhatsApp Image 2022-06-07 at 11 21 21 AM
WhatsApp Image 2022-06-07 at 11 44 56 AM

### CODE REVIEW

File: app/src/main/java/com/example/practiceapp/LoginActivity.kt [hunk 1]
```diff
@@ -23,25 +23,24 @@ class LoginActivity : AppCompatActivity() {
...
-                }else{
-                    Toast.makeText(this, " Usuario o contraseña incorrecto", Toast.LENGTH_SHORT).show()
+            if (emaill.isEmpty() || pass.isEmpty()) {
+                Toast.makeText(this, "Debes ingresar ambos campos para iniciar sesion", Toast.LENGTH_SHORT).show()
+            } else {
+                val db = (application as UserApp).database
+                val userdb = db.userDao().getById(emaill)
+                if (userdb != null) {
+                    val passdb = userdb.password
+                    if (passdb == pass) {
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/LoginActivity.kt]: you can avoid this validation by having a method and query in the DAO that searches by email and password instead of only searching by email.

File: app/src/main/java/com/example/practiceapp/UserFragment.kt [hunk 2]
```diff
@@ -5,55 +5,26 @@ import androidx.fragment.app.Fragment
...
-        return inflater.inflate(R.layout.fragment_user, container, false)
-    }
+        super.onCreate(savedInstanceState)
+        val v = inflater.inflate(R.layout.fragment_user, container, false)
+        val recyclerView = v.findViewById<RecyclerView>(R.id.frag_user_recyclerView)
+        val manager = LinearLayoutManager(context)
+        val decoration = DividerItemDecoration(context,manager.orientation)
+        recyclerView.layoutManager = manager
+        recyclerView.adapter = FilmAdapter(FilmsRecyclerviewProvider.films)
+        recyclerView.addItemDecoration(decoration)
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/UserFragment.kt]: this code can be moved away from this state to a more proper one, check the fragment lifecycle to identify which one suits better.

File: app/src/main/res/layout/fragment_home.xml [hunk 3]
```diff
@@ -6,15 +6,26 @@
...
-        app:layout_constraintBottom_toBottomOf="parent"/>
+        app:layout_constraintTop_toTopOf="parent"
+        >
+
+        <com.google.android.material.imageview.ShapeableImageView
+            android:layout_width="match_parent"
+            android:layout_height="250dp"
+            android:background="@drawable/ic_cdnlogo_com_imdbnew" />
+
+        <com.google.android.material.imageview.ShapeableImageView
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_home.xml]: why two shapeableImageView with the same background? and why inside a FrameLayout?

File: app/src/main/res/layout/fragment_user.xml [hunk 4]
```diff
@@ -1,24 +1,200 @@
...
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        android:background="@color/white"
+        android:layout_height="match_parent"
+        tools:context=".UserFragment">
+
+
+        <LinearLayout
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: if you're already using a ConstraintLayout above this, you don't need the LinearLayout and can achieve the same result by using the proper constraints, plus avoid nesting layouts.

File: app/src/main/res/layout/fragment_user.xml [hunk 5]
```diff
@@ -1,24 +1,200 @@
...
+            android:layout_marginTop="@dimen/medium_size"
+            android:background="@color/buscar_edit_text"
+            app:layout_constraintTop_toBottomOf="@id/layout_user"
+            app:layout_constraintEnd_toEndOf="@id/layout_user"
+            app:layout_constraintStart_toStartOf="@id/layout_user"/>
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/frag_user_recyclerView"
+            android:layout_width="0dp"
+            android:layout_height="140dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: you're already using the dimens file for other specs, please continue with that good practice and avoid hardcoding sizes

File: app/src/main/res/layout/fragment_user.xml [hunk 6]
```diff
@@ -1,24 +1,200 @@
...
+            app:layout_constraintStart_toStartOf="@+id/frag_user_recyclerView"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_tv_blanc4"/>
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/frag_user_tv_list_follow_description"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/medium_size"
+            android:text="Crea una lista de seguimiento para no perderte ninguna pelicula o serie de tv"
+            android:textSize="15dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: text sizes are defined in sp instead of dp, and you can also define them in the dimens file

File: app/src/main/res/layout/fragment_user.xml [hunk 7]
```diff
@@ -1,24 +1,200 @@
...
+            android:textStyle="bold"
+            app:layout_constraintStart_toStartOf="@+id/frag_user_recyclerView"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_tv_blanc4"/>
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/frag_user_tv_list_follow_description"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/medium_size"
+            android:text="Crea una lista de seguimiento para no perderte ninguna pelicula o serie de tv"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: just as you're currently doing for some dimens, hardcoded strings also go into the strings.xml file

File: app/src/main/res/layout/fragment_user.xml [hunk 8]
```diff
@@ -1,24 +1,200 @@
...
+            />
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/frag_user_tv_list_follow_bottom"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:background="@drawable/style_border"
+            android:backgroundTint="@color/colorApp"
+            android:layout_marginTop="@dimen/medium_size"
+            android:textAppearance="@style/TextAppearance.AppCompat.Large"
+            android:textSize="18sp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: in this case you're using the right dimension spec (sp) but please move it to the dimes file

File: app/src/main/res/layout/fragment_user.xml [hunk 9]
```diff
@@ -1,24 +1,200 @@
...
+            android:text="Empieza una lista de seguimiento"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: move the texts to the strings file

File: app/src/main/res/layout/fragment_user.xml [hunk 10]
```diff
@@ -1,24 +1,200 @@
...
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/medium_size"
+            android:text="No has visitado ninguna pagina recientemente"
+            android:textSize="15dp"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_tv_recent_views"
+            app:layout_constraintEnd_toEndOf="@id/frag_user_tv_list_follow_bottom"
+            app:layout_constraintStart_toStartOf="@id/frag_user_tv_list_follow_bottom"
+            />
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/frag_user_tv_blanc3"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: what are blanc2 and blanc3 for?

**Review by jrojas25 — CHANGES_REQUESTED** [review]
In addition to the changes explicitly requested on each file please make sure all files fulfill the same requirements, also check the imported image that overrode the app icon, have in mind that this hardcoded image for the movies don't have to be and SVG and can be used as png.

**Comment by deliasgomez** [comment]
WhatsApp Image 2022-06-13 at 6 44 19 PM
WhatsApp Image 2022-06-13 at 6 44 28 PM

File: app/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -14,11 +14,11 @@
 
 
         <activity
-            android:name=".BuscarActivity"
+            android:name=".SearchActivity"
             android:exported="false"
             android:theme="@style/buscar_activity"/>
         <activity
-            android:name=".ActivityRegistro"
+            android:name=".ActivityRegister"
```
Comment (jrojas25) [inline app/src/main/AndroidManifest.xml]: you have SearchActivity and ActivityRegister, please standardize the naming by having `Activity` as either prefix or suffix but not both.

File: app/src/main/java/com/example/practiceapp/Database/UserDao.kt [hunk 12]
```diff
@@ -12,4 +12,7 @@ interface  UserDao {
     @Insert
     fun insert(user: User)
 
+    @Query("SELECT * FROM user WHERE email = :email and password = :password")
+    fun getByIDAnDPass(email: String, password : String ): User
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/Database/UserDao.kt]: small typo in `getByIDAnDPass`, should be `getByIdAndPass`

File: app/src/main/java/com/example/practiceapp/FilmAdapter.kt [hunk 13]
```diff
@@ -8,7 +8,7 @@ class FilmAdapter(private val filmsList: List<FilmsRecyclerviewSearch>) : Recycl
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FilmViewHolder {
         val view = LayoutInflater.from(parent.context)
-            .inflate(R.layout.item_film, parent, false)
+            .inflate(R.layout.view_recycler_frag_search, parent, false)
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/FilmAdapter.kt]: the name of the layout is not clear, are you inflating a recycler from an Adapter?

File: app/src/main/java/com/example/practiceapp/HomeFragment.kt [hunk 14]
```diff
@@ -1,59 +1,34 @@
...
-                arguments = Bundle().apply {
-                    putString(ARG_PARAM1, param1)
-                    putString(ARG_PARAM2, param2)
-                }
-            }
+    @SuppressLint("WrongConstant")
+    override fun onViewCreated(itemView: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(itemView, savedInstanceState)
+
+        val recyclerView = view?.findViewById<RecyclerView>(R.id.frag_home_recycler_view)
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/HomeFragment.kt]: now that you have achieved the purpose of making this screen work, please use viewBinding to get the view elements

File: app/src/main/java/com/example/practiceapp/HomeFragment.kt [hunk 15]
```diff
@@ -1,59 +1,34 @@
...
-        // TODO: Rename and change types and number of parameters
-        @JvmStatic
-        fun newInstance(param1: String, param2: String) =
-            HomeFragment().apply {
-                arguments = Bundle().apply {
-                    putString(ARG_PARAM1, param1)
-                    putString(ARG_PARAM2, param2)
-                }
-            }
+    @SuppressLint("WrongConstant")
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/HomeFragment.kt]: don't use this and fix the warning instead

File: app/src/main/java/com/example/practiceapp/LoginActivity.kt [hunk 16]
```diff
@@ -19,47 +19,42 @@ class LoginActivity : AppCompatActivity() {
...
                 }
             }
         })
-        binding.loginEtpass.addTextChangedListener(object : TextWatcher{
+        binding.loginEtPass.addTextChangedListener(object : TextWatcher{
             override fun afterTextChanged(s: Editable?) {}
             override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
             override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
-                if (binding.loginEtUser.text?.length!! > 0 && binding.loginEtpass.text?.length!! > 0 ){
+                if (binding.loginEtUser.text?.length!! > 0 && binding.loginEtPass.text?.length!! > 0 ){
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/LoginActivity.kt]: Lines 46 and 57 have the same logic, you can move it to a function and call it from both places to avoid repeating code

Comment (jrojas25) [reply app/src/main/java/com/example/practiceapp/LoginActivity.kt]: and same for `RegisterActivity`, now as this validation is required in two different classes you can have a Utils class with this validation as a function and then calling it from different places

File: app/src/main/java/com/example/practiceapp/RecyclerViewFragHomeProvider.kt [hunk 17]
```diff
@@ -0,0 +1,13 @@
+package com.example.practiceapp
+
+class RecyclerViewFragHomeProvider {
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/RecyclerViewFragHomeProvider.kt]: the naming is not clear, why is this related to REcyclerViewFragHome? you should have only one list that I think you already have in another provider

File: app/src/main/java/com/example/practiceapp/SearchFragment.kt [hunk 18]
```diff
@@ -8,23 +9,26 @@ import android.view.ViewGroup
...
-        super.onCreate(savedInstanceState)
-        val v = inflater.inflate(R.layout.fragment_search, container, false)
-        val recyclerView = v.findViewById<RecyclerView>(R.id.recyclerFilms)
+        return inflater.inflate(R.layout.fragment_search, container, false)
+    }
+
+    override fun onViewCreated(itemView: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(itemView, savedInstanceState)
+
+        val recyclerView = view?.findViewById<RecyclerView>(R.id.recyclerFilms)
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/SearchFragment.kt]: please use viewBinding

File: app/src/main/java/com/example/practiceapp/UserFragment.kt [hunk 19]
```diff
@@ -1,59 +1,37 @@
...
-                arguments = Bundle().apply {
-                    putString(ARG_PARAM1, param1)
-                    putString(ARG_PARAM2, param2)
-                }
-            }
+    @SuppressLint("WrongConstant")
+    override fun onViewCreated(itemView: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(itemView, savedInstanceState)
+
+        val recyclerView = view?.findViewById<RecyclerView>(R.id.frag_user_recyclerView)
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/UserFragment.kt]: please use viewBinding

File: app/src/main/res/drawable/yellow_bar.xml [hunk 20]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="rectangle">
+
+    <solid android:color="@color/colorApp"/>
+    <corners android:radius="@dimen/size_10"/>
+    <size android:width="@dimen/size_5"
+        android:height="25dp"/>
```
Comment (jrojas25) [inline app/src/main/res/drawable/yellow_bar.xml]: you're already using the dimens file for the width, why not using it for height?

File: app/src/main/res/layout/activity_login.xml [hunk 21]
```diff
@@ -29,162 +29,179 @@
...
             android:inputType="textWebPassword"
+            android:textSize="@dimen/size_text_20"
+            android:layout_marginStart="@dimen/size_40"
+            android:layout_marginEnd="@dimen/size_40"
             app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintHorizontal_bias="0.0"
-            app:layout_constraintStart_toStartOf="@+id/loginTvpass"
-            app:layout_constraintTop_toBottomOf="@+id/loginTvpass"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/loginTvPass"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_login.xml]: when referencing existing ids avoid using the `+`, so this will also work as `"@id/loginTvPass"`

File: app/src/main/res/values/strings.xml [hunk 22]
```diff
@@ -7,30 +7,41 @@
...
     <string name="login_tv_pass">Contraseña</string>
-    <string name="login_tv_forgotten_pass"> </string>
+    <string name="login_tv_forgotten_pass">¿Olvidaste tu contraseña?</string>
     <string name="login_other_options">Ó podes ingresar con</string>
-    <string name="login_botton_face">TODO</string>
-    <string name="login_botton_apple">TODO</string>
-    <string name="login_botton_google">TODO</string>
+    <string name="login_button_face">TODO</string>
+    <string name="login_button_apple">TODO</string>
+    <string name="login_button_google">TODO</string>
```
Comment (jrojas25) [inline app/src/main/res/values/strings.xml]: what is this for?

**Review by jrojas25 — CHANGES_REQUESTED** [review]
Great job using both the dimens and strings files in your layout.

There still a couple things to fix and also make sure you revert the changes made to app/src/main/ic_launcher-playstore.png

**Comment by deliasgomez** [comment]
WhatsApp Image 2022-06-15 at 8 04 00 AM

File: app/src/main/java/com/example/practiceapp/HomeFragment.kt [hunk 23]
```diff
@@ -5,55 +5,27 @@ import androidx.fragment.app.Fragment
...
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        arguments?.let {
-            param1 = it.getString(ARG_PARAM1)
-            param2 = it.getString(ARG_PARAM2)
-        }
-    }
-
+    private var binding : FragmentHomeBinding? = null
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/HomeFragment.kt]: please follow the guidelines to use binding from fragments

File: app/src/main/java/com/example/practiceapp/RegisterActivity.kt [hunk 24]
```diff
@@ -31,48 +30,49 @@ class ActivityRegistro : AppCompatActivity() {
...
             val db = (application as UserApp).database
             try {
-                db.userDao().insert(User(name,email,password))
-                Toast.makeText(this,"Usuario agregado",Toast.LENGTH_SHORT).show()
-                startActivity(Intent(this,LoginActivity::class.java))
-            }catch (e: SQLiteConstraintException){
-                Toast.makeText(this,"El usuario ya existe",Toast.LENGTH_SHORT).show()
+                db.userDao().insert(User(name, email, password))
+                Toast.makeText(this, "Usuario agregado", Toast.LENGTH_SHORT).show()
+                startActivity(Intent(this, LoginActivity::class.java))
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/RegisterActivity.kt:73]: it is a good practice to finish the current Activity when leaving the flow

File: app/src/main/java/com/example/practiceapp/SearchFragment.kt [hunk 25]
```diff
@@ -1,31 +1,30 @@
...
+    private var binding : FragmentSearchBinding? = null
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        super.onCreate(savedInstanceState)
-        val v = inflater.inflate(R.layout.fragment_search, container, false)
-        val recyclerView = v.findViewById<RecyclerView>(R.id.recyclerFilms)
+    ): View {
+        binding = FragmentSearchBinding.inflate(inflater,container,false)
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/SearchFragment.kt]: please follow the guidelines to use binding for fragments

File: app/src/main/java/com/example/practiceapp/UserApp.kt [hunk 26]
```diff
@@ -6,5 +6,7 @@ import com.example.practiceapp.Database.UserDataBase
 
 class UserApp : Application() {
     val database: UserDataBase by lazy { UserDataBase.getDatabase(this) }
+    fun hola(){
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/UserApp.kt]: what's this function for?

File: app/src/main/java/com/example/practiceapp/UserFragment.kt [hunk 27]
```diff
@@ -1,59 +1,32 @@
...
-
+    private var binding : FragmentUserBinding? = null
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_user, container, false)
+    ): View {
+        binding = FragmentUserBinding.inflate(inflater, container, false)
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/UserFragment.kt]: please follow the guidelines to use binding for fragments

File: app/src/main/java/com/example/practiceapp/Utils.kt [hunk 28]
```diff
@@ -0,0 +1,22 @@
...
+
+class Utils {
+    fun fullFieldsLogin(view1 : EditText, view2 : EditText,bottom : Button){
+        if (view1.text?.length!! > 0 && view2.text?.length!! > 0 ){
+            bottom.setBackgroundColor(Color.parseColor("#4B4747"))
+        }else{
+            bottom.setBackgroundColor(Color.parseColor("#9D9C9C"))
+        }
+    }
+    fun fullFieldsregister(view1: EditText,view2: EditText,view3: EditText,bottom: Button){
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/Utils.kt]: typo, register should start with capital R in the method name

File: app/src/main/res/layout/activity_login.xml [hunk 29]
```diff
@@ -29,162 +29,179 @@
...
+            app:layout_constraintStart_toStartOf="@id/loginTvUser"
+            app:layout_constraintTop_toBottomOf="@id/loginTvUser"
             tools:ignore="SpeakableTextPresentCheck" />
 
         <com.google.android.material.textview.MaterialTextView
-            android:id="@+id/loginTvpass"
+            android:id="@+id/loginTvPass"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_marginTop="10dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_login.xml]: use dimens

File: app/src/main/res/layout/activity_login.xml [hunk 30]
```diff
@@ -29,162 +29,179 @@
...
-            app:layout_constraintTop_toBottomOf="@+id/loginTvpass"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/loginTvPass"
             tools:ignore="SpeakableTextPresentCheck" />
 
         <com.google.android.material.textview.MaterialTextView
             android:id="@+id/loginTvForget"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_marginTop="3dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_login.xml]: use dimens

File: app/src/main/res/layout/activity_login.xml [hunk 31]
```diff
@@ -29,162 +29,179 @@
...
 
         <com.google.android.material.floatingactionbutton.FloatingActionButton
-            android:id="@+id/loginFabgoogle"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            android:id="@+id/loginFabGoogle"
+            android:layout_width="@dimen/size_60"
+            android:layout_height="@dimen/size_60"
             android:layout_margin="16dp"
             android:layout_marginStart="12dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_login.xml]: use dimens

File: app/src/main/res/layout/activity_login.xml [hunk 32]
```diff
@@ -29,162 +29,179 @@
...
-            android:contentDescription="@string/login_botton_google" />
+            android:contentDescription="@string/login_image_google_description"/>
 
         <com.google.android.material.floatingactionbutton.FloatingActionButton
             android:id="@+id/loginFabFacebook"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            android:layout_width="@dimen/size_60"
+            android:layout_height="@dimen/size_60"
             android:layout_margin="16dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_login.xml]: use dimens

File: app/src/main/res/layout/fragment_home.xml [hunk 33]
```diff
@@ -5,16 +5,95 @@
...
+        android:layout_marginTop="@dimen/size_60"
+        android:textSize="@dimen/size_text_20"
+        android:textStyle="bold"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/frag_home_secondary_image"/>
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/frag_home_recycler_view"
+        android:layout_width="0dp"
+        android:layout_height="250dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_home.xml:82]: use dimens

File: app/src/main/res/layout/fragment_user.xml [hunk 34]
```diff
@@ -1,24 +1,193 @@
...
+            android:backgroundTint="@color/black"
+            android:background="@drawable/vector_profile_user"
+            android:layout_marginStart="@dimen/size_20"
+            android:layout_marginTop="@dimen/size_20"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintStart_toStartOf="parent"/>
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/frag_user_user_name"
+            android:layout_width="0dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: why a 0dp width?

File: app/src/main/res/layout/fragment_user.xml [hunk 35]
```diff
@@ -1,24 +1,193 @@
...
+            android:layout_marginStart="@dimen/size_20"
+            android:layout_marginEnd="@dimen/size_20"
+            android:layout_marginTop="@dimen/size_10"
+            android:background="@color/search_edit_text"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_user_photo"
+            app:layout_constraintStart_toStartOf="parent" />
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/frag_user_recyclerView"
+            android:layout_width="0dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: why a 0dp width?

File: app/src/main/res/layout/fragment_user.xml [hunk 36]
```diff
@@ -1,24 +1,193 @@
...
+            android:textColor="@color/black"
+            android:layout_marginTop="@dimen/size_60"
+            android:textSize="@dimen/size_text_20"
+            android:textStyle="bold"
+            app:layout_constraintStart_toStartOf="@id/frag_user_recyclerView"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_recyclerView"/>
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/frag_user_tv_list_follow_description"
+            android:layout_width="0dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: why a 0dp width?

File: app/src/main/res/layout/fragment_user.xml [hunk 37]
```diff
@@ -1,24 +1,193 @@
...
+            android:layout_marginTop="@dimen/size_10"
+            android:text="@string/frag_user_follow_list_description"
+            android:textSize="@dimen/size_text_15"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_tv_list_follow"
+            app:layout_constraintEnd_toEndOf="@id/frag_user_recyclerView"
+            app:layout_constraintStart_toStartOf="@id/frag_user_recyclerView" />
+
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/frag_user_tv_list_follow_bottom"
+            android:layout_width="0dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: why a 0dp width?

File: app/src/main/res/layout/fragment_user.xml [hunk 38]
```diff
@@ -1,24 +1,193 @@
...
+            android:layout_marginTop="@dimen/size_60"
+            android:textColor="@color/black"
+            android:textSize="@dimen/size_text_20"
+            android:textStyle="bold"
+            app:layout_constraintStart_toStartOf="@id/frag_user_tv_list_follow_bottom"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_tv_list_follow_bottom"/>
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/frag_user_tv_recent_views_description"
+            android:layout_width="0dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: why a 0dp width?

File: app/src/main/res/layout/fragment_user.xml [hunk 39]
```diff
@@ -1,24 +1,193 @@
...
+            android:layout_marginTop="@dimen/size_60"
+            android:textColor="@color/black"
+            android:textSize="@dimen/size_text_20"
+            android:textStyle="bold"
+            app:layout_constraintStart_toStartOf="@id/frag_user_tv_list_follow_bottom"
+            app:layout_constraintTop_toBottomOf="@id/frag_user_tv_recent_views_description"/>
 
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/frag_user_tv_favorite_persons_description"
+            android:layout_width="0dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_user.xml]: why a 0dp width?

**Review by jrojas25 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/practiceapp/CardsViewHolder.kt [hunk 40]
```diff
@@ -0,0 +1,16 @@
...
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class CardsViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    private val function = view.findViewById<TextView>(R.id.card_functions)
+    private val resumeFunctions = view.findViewById<TextView>(R.id.card_resume_functions)
+    private val totalFunctionsPerformed = view.findViewById<TextView>(R.id.card_total_functions_performed)
+    fun render(cardsModel: Cards) {
+        function.text = cardsModel.name
+        resumeFunctions.text = cardsModel.function
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/CardsViewHolder.kt]: why `function` uses `cardsModel.name` and  `resumeFunctions` uses `cardsModel.function`? the names are not matching

**Review by jrojas25 — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_login.xml (hunk 31 above)
Comment (deliasgomez) [reply app/src/main/res/layout/activity_login.xml]: ok

File: app/src/main/AndroidManifest.xml (hunk 11 above)
Comment (deliasgomez) [reply app/src/main/AndroidManifest.xml]: Thanks for the advice

File: app/src/main/java/com/example/practiceapp/RegisterActivity.kt (hunk 24 above)
Comment (jrojas25) [reply app/src/main/java/com/example/practiceapp/RegisterActivity.kt:73]: Remember to only resolve comments with either a code change or by giving your opinion regarding the changes made and when the reviewer agrees with that.

In this case what I'm looking for with this comment is for you to add a `finish()` call after starting the LoginActivity

File: app/src/main/java/com/example/practiceapp/SearchActivity.kt [hunk 41]
```diff
@@ -3,21 +3,22 @@ package com.example.practiceapp
...
 import android.os.Bundle
 import androidx.fragment.app.Fragment
-import com.example.practiceapp.databinding.ActivityBuscarBinding
+import com.example.practiceapp.databinding.ActivitySearchBinding
+
 private lateinit var currentFragment : Fragment
-private lateinit var binding: ActivityBuscarBinding
-class BuscarActivity : AppCompatActivity() {
+private lateinit var binding: ActivitySearchBinding
+class SearchActivity : AppCompatActivity() {
```
Comment (jrojas25) [inline app/src/main/java/com/example/practiceapp/SearchActivity.kt]: Why is this Activity called `SearchActivity`?

**Review by jrojas25 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/practiceapp/SearchActivity.kt (hunk 41 above)
Comment (deliasgomez) [reply app/src/main/java/com/example/practiceapp/SearchActivity.kt]: Thanks for the advice.

**Review by jrojas25 — APPROVED** [review]
Good to merge
